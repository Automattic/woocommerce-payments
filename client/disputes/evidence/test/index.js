/** @format */
/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { DisputeEvidenceForm } from '../';

describe( 'Dispute evidence form', () => {
	test( 'needing response, renders correctly', () => {
        const dispute = {
            id: 'dp_asdfghjkl',
            amount: 1000,
            created: 1572590800,
            evidence: {
                customer_purchase_ip: '127.0.0.1',
                uncategorized_text: '',
            },
            evidence_details: {
                due_by: 1573199200,
            },
            reason: 'fraudulent',
            status: 'needs_response',
        };

		const form = shallow(
			<DisputeEvidenceForm
                evidence={ dispute.evidence }
                showPlaceholder={ false }
                // onChange
                // onSave
                readOnly={ false }
			/>
		);
		expect( form ).toMatchSnapshot();
    } );

	test( 'not needing response, renders correctly', () => {
        const dispute = {
            id: 'dp_zxcvbnm',
            amount: 1050,
            created: 1572480800,
            evidence: {
                customer_purchase_ip: '127.0.0.1',
                uncategorized_text: 'winning_evidence',
            },
            evidence_details: {
                due_by: 1573099200,
            },
            reason: 'general',
            status: 'under_review',
        };

		const form = shallow(
			<DisputeEvidenceForm
                evidence={ dispute.evidence }
                showPlaceholder={ false }
                // onChange
                // onSave
                readOnly={ false }
			/>
		);
		expect( form ).toMatchSnapshot();
	} );
} );
