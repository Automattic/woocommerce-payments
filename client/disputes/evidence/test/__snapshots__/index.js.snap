// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispute evidence form needing response, renders correctly 1`] = `
<div>
  <div
    class="components-card is-size-large css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-large e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      General evidence
    </div>
    <div
      class="components-card__body is-size-large css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="inspector-textarea-control-0"
          >
            Product description
          </label>
          <textarea
            class="components-textarea-control__input css-gw3wbu-StyledTextarea-inputStyleNeutral-inputStyleFocus-inputControl ebk7yr50"
            id="inspector-textarea-control-0"
            rows="4"
          />
        </div>
      </div>
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="inspector-text-control-0"
          >
            Customer name
          </label>
          <input
            class="components-text-control__input"
            id="inspector-text-control-0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="form-file-upload-base-control-customer_signature"
          >
            Customer signature
          </label>
          <div
            class="components-drop-zone__provider"
          >
            <div
              class="components-drop-zone"
            />
          </div>
          <div
            class="file-upload"
          >
            <div
              class="components-form-file-upload"
            >
              <button
                class="components-button is-secondary has-text has-icon"
                id="form-file-upload-customer_signature"
                type="button"
              >
                <svg
                  class="gridicon gridicons-add-outline"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm5 9h-4V7h-2v4H7v2h4v4h2v-4h4v-2z"
                    />
                  </g>
                </svg>
                Upload file
              </button>
              <input
                accept=".pdf, image/png, image/jpeg"
                style="display: none;"
                type="file"
              />
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="inspector-text-control-1"
          >
            Service date
          </label>
          <input
            class="components-text-control__input"
            id="inspector-text-control-1"
            max="2021-06-24"
            type="date"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="components-card is-size-large css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-large css-xmjzce-BodyUI e1q7k77g3"
    >
      <p>
        When you submit your evidence, we'll format it and send it to the cardholder's bank, then email you once the dispute has been decided.
      </p>
      <p>
        <strong>
          Evidence submission is final.
        </strong>
         
        You can also save this evidence for later instead of submitting it immediately.
         
        <strong>
          We will automatically submit any saved evidence at the due date.
        </strong>
      </p>
    </div>
    <div
      class="components-flex components-card__footer is-size-large e1q7k77g4 css-oliaol-Flex-FooterUI eboqfv50"
    >
      <div>
        <button
          class="components-button is-primary"
          type="button"
        >
          Submit evidence
        </button>
        <button
          class="components-button is-secondary"
          type="button"
        >
          Save for later
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dispute evidence form not needing response, renders correctly 1`] = `
<div>
  <div
    class="components-card is-size-large css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-large e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      General evidence
    </div>
    <div
      class="components-card__body is-size-large css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="inspector-textarea-control-1"
          >
            Product description
          </label>
          <textarea
            class="components-textarea-control__input css-gw3wbu-StyledTextarea-inputStyleNeutral-inputStyleFocus-inputControl ebk7yr50"
            disabled=""
            id="inspector-textarea-control-1"
            rows="4"
          />
        </div>
      </div>
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="inspector-text-control-2"
          >
            Customer name
          </label>
          <input
            class="components-text-control__input"
            disabled=""
            id="inspector-text-control-2"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="form-file-upload-base-control-customer_signature"
          >
            Customer signature
          </label>
          <div
            class="components-drop-zone__provider"
          >
            <div
              class="components-drop-zone"
            />
          </div>
          <div
            class="file-upload"
          >
            <div
              class="components-form-file-upload"
            >
              <button
                class="components-button is-secondary has-text has-icon"
                disabled=""
                id="form-file-upload-customer_signature"
                type="button"
              >
                <svg
                  class="gridicon gridicons-add-outline"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm5 9h-4V7h-2v4H7v2h4v4h2v-4h4v-2z"
                    />
                  </g>
                </svg>
                Upload file
              </button>
              <input
                accept=".pdf, image/png, image/jpeg"
                style="display: none;"
                type="file"
              />
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
      >
        <div
          class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
        >
          <label
            class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
            for="inspector-text-control-3"
          >
            Service date
          </label>
          <input
            class="components-text-control__input"
            disabled=""
            id="inspector-text-control-3"
            max="2021-06-24"
            type="date"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dispute evidence page renders correctly 1`] = `
<div>
  <div
    class="wcpay-dispute-evidence woocommerce-payments-page is-narrow"
  >
    <div
      class="components-card is-size-large css-1xs3c37-CardUI e1q7k77g0"
    >
      <div
        class="components-flex components-card__header is-size-large e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
      >
        Challenge dispute
      </div>
      <div
        class="components-card__body is-size-large css-xmjzce-BodyUI e1q7k77g3"
      >
        <div
          class="wcpay-dispute-info"
        >
          <div
            class="wcpay-dispute-info-item"
          >
            <span
              class="wcpay-dispute-info-key"
            >
              Dispute date: 
            </span>
            <span
              class="wcpay-dispute-info-value"
            >
              Nov 1, 2019
            </span>
          </div>
          <div
            class="wcpay-dispute-info-item"
          >
            <span
              class="wcpay-dispute-info-key"
            >
              Disputed amount: 
            </span>
            <span
              class="wcpay-dispute-info-value"
            >
              $10.00 USD
            </span>
          </div>
          <div
            class="wcpay-dispute-info-item"
          >
            <span
              class="wcpay-dispute-info-key"
            >
              Respond by: 
            </span>
            <span
              class="wcpay-dispute-info-value"
            >
              Nov 8, 2019 - 2:46AM
            </span>
          </div>
          <div
            class="wcpay-dispute-info-item"
          >
            <span
              class="wcpay-dispute-info-key"
            >
              Reason: 
            </span>
            <span
              class="wcpay-dispute-info-value"
            >
              Fraudulent
            </span>
          </div>
          <div
            class="wcpay-dispute-info-item"
          >
            <span
              class="wcpay-dispute-info-key"
            >
              Transaction ID: 
            </span>
            <span
              class="wcpay-dispute-info-value"
            >
              <a
                data-link-type="wc-admin"
                href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions%2Fdetails"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="components-card is-size-large css-1xs3c37-CardUI e1q7k77g0"
    >
      <div
        class="components-flex components-card__header is-size-large e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
      >
        Product type
      </div>
      <div
        class="components-card__body is-size-large css-xmjzce-BodyUI e1q7k77g3"
      >
        <div
          class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
        >
          <div
            class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
          >
            <div
              class="components-flex components-select-control e1cr7zh10 css-38uqbv-Flex-Root eboqfv50"
            >
              <div
                class="components-flex__item e1cr7zh14 css-d373l0-Item-LabelWrapper eboqfv51"
              />
              <div
                class="components-input-control__container css-1ih9r4d-Container e1cr7zh11"
              >
                <select
                  class="components-select-control__input css-1hft3f5-Select e12x0a390"
                  data-testid="dispute-challenge-product-type-selector"
                  id="inspector-select-control-0"
                >
                  <option
                    disabled=""
                    value=""
                  >
                    Select one…
                  </option>
                  <option
                    value="physical_product"
                  >
                    Physical product
                  </option>
                  <option
                    value="digital_product_or_service"
                  >
                    Digital product or service
                  </option>
                  <option
                    value="offline_service"
                  >
                    Offline service
                  </option>
                  <option
                    value="multiple"
                  >
                    Multiple product types
                  </option>
                </select>
                <span
                  class="components-input-control__suffix css-1lym30p-Suffix e1cr7zh17"
                >
                  <div
                    class="css-163e8ky-DownArrowWrapper e12x0a391"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      height="18"
                      role="img"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
                      />
                    </svg>
                  </div>
                </span>
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-96sjau-BackdropUI e1cr7zh15"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
